src/averageAges.js

```diff
@@ -37,7 +44,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const mothers = withChildren
+    ? women.filter(woman =>
+      people.some(person => person.mother === woman.name))
+    : women;
+
+  const sumOfAge = mothers.reduce((sum, person) =>
+    sum + (person.died - person.born), 0);
+
+  return sumOfAge / mothers.length;
```

Comment:
you have the same logic in men function, please create a function helper for calculating average age